torch/nn/parallel/distributed.py

```diff
@@ -217,6 +217,11 @@ class DistributedDataParallel(Module):
                          are getting different gradients, which should not
                          happen if DistributedDataParallel is correctly used.
                          (default: ``False``)
+        scatter_single_device:  when set to ``False``, scatter functionality is
```

Comment:
If we have to expose this flag, would it be clear if we jus call it `scatter_to_devices`, which governs scatter or not regardless the value of `device_ids`? Then the condition in forward:

```python
if self.device_ids and (len(self.device_ids) > 1 or self.scatter_single_device):
```

Becomes

```python
if self.device_ids and self.scatter_to_devices:
```

Will the above miss any use cases?

Reply:
This sounds simpler to me!  Give the user the option to manually handle the scatter even for multi-GPU.  I like it  :)

Reply:
Actually, I'm not sure how that would work to use multi-GPU per process and not scatter the data inside the model.  I just don't have experience with multi-GPU per process, so I was trying to avoid changing anything to do with that.

Reply:
Ahh, you are right, how about this?

```python
if self.device_ids:
    if self.scatter_to_devices:
        inputs, kwargs = self.scatter(inputs, kwargs, self.device_ids)
    if len(self.device_ids) == 1:
        output = self.module(*inputs[0], **kwargs[0])
    else:
        outputs = self.parallel_apply(self._module_copies[:len(inputs)], inputs, kwargs)
        output = self.gather(outputs, self.output_device)
else:
```

Reply:
In the case of no scattering and single device, that would still execute the problematic line:
```python
output = self.module(*inputs[0], **kwargs[0])
```
which needs to be:
```python
output = self.module(*inputs, **kwargs)
```
like in the PR. :)

Reply:
Could keep the PR logic but change the name to `scatter_to_device` if that's more clear?

Reply:
Yes, let do that for now.

Reply:
If you could please make changes to the current PR I will land it. Thanks for contributing!

Reply:
cool, done!  thanks for accepting :)